core/primitives/src/trie_key.rs

```diff
@@ -38,7 +38,10 @@ pub mod col {
     /// NOTE: It is a singleton per shard.
     pub const DELAYED_RECEIPT_INDICES: u8 = 7;
     /// This column id is used when storing delayed receipts, because the shard is overwhelmed.
-    pub const DELAYED_RECEIPT: u8 = 8;
+    /// Here we share the identifier with `DELAYED_RECEIPT_INDICES` for historical reasons. It
+    /// is valid because the length of `TrieKey::DelayedReceipt` is always greater than
+    /// `TrieKey::DelayedReceiptIndices` when serialized to bytes.
+    pub const DELAYED_RECEIPT: u8 = DELAYED_RECEIPT_INDICES;
```

Comment:
Is this statically safe? E.i. in the match case above, would the compiler produce an error without the disambiguatinig if statements?

Reply:
Yeah, this is actually safe and compiler detects the same value. Though now I realised that it is probably still confusing, so I will change it to have a single `DELAYED_RECEIPT_AND_INDICES` constant.

Reply:
@wacban done, please take a look